### PR TITLE
go/oasis-test-runner: Refactor code to refer to scenario(s) consistently

### PR DESCRIPTION
**Pull request by tjanez**
go/oasis-test-runner/cmd: Rename `--test` flag to `--scenario` flag (`-s` for short version) for consistency.

go/oasis-test-runner/env:
- Rename `TestInstanceInfo` type to `ScenarioInstanceInfo` and its `Test` field to `Scenario` field for consistency.
- Rename `Env`'s `TestInfo()` and `WriteTestInfo()` methods to `ScenarioInfo()` and `WriteScenarioInfo()` for consistency.

go/oasis-node/cmd/common/metrics: Use "scenario" metrics label instead of "test" for consistency.

### CODE REVIEW

**Comment by ptrus** [comment]
Also fix: https://github.com/oasisprotocol/oasis-core/blob/0466b12b6c3ab9a395bcc748006d442869737242/.buildkite/scripts/daily_txsource.sh#L9

File: .changelog/3108.breaking.1.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+go/oasis-test-runner/cmd: Rename `--test` flag to `--scenario` flag
```
Comment (kostko) [inline .changelog/3108.breaking.1.md:1]: Make sure to update all the CI pipelines, I don't see any here and I know there are places that use `-t`.

Comment (tjanez) [reply .changelog/3108.breaking.1.md:1]: Thanks for the pointer, I think I fixed all these.

**Comment by tjanez** [comment]
> Also fix:
> 
> https://github.com/oasisprotocol/oasis-core/blob/0466b12b6c3ab9a395bcc748006d442869737242/.buildkite/scripts/daily_txsource.sh#L9

Done.

**Pull request merged**